snippets: add snippet for tfm logging on shared uart (#12488)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf91/nrf9160.rst

See :ref:`cmake_options` for more details.

.. _tfm_enable_share_uart:

> Shared UART for application and TFM logging
> ============================================

Reviewer: ```suggestion
Shared UART for application and TF-M logging
=============================================
```

See :ref:`cmake_options` for more details.

.. _tfm_enable_share_uart:

> Shared UART for application and TFM logging
> ===========================================

Reviewer: ```suggestion
Shared UART for application and TF-M logging
============================================
```

> If you want to activate TF-M logging while having modem traces enabled, it can be useful to direct the TF-M logs to the UART (UART0) used by the application.

Reviewer: ```suggestion
If you want to activate TF-M logging while having modem traces enabled, it can be useful to direct the TF-M logs to the UART (**UART0**) used by the application.
```

To activate both modem traces and TF-M logs, use the following command:

.. parsed-literal::
   :class: highlight

>    west build --board *your_board* -S nrf91-modem-trace-uart -S tfm-enable-share-uart

Reviewer: ```suggestion
   west build --board *build target* -S nrf91-modem-trace-uart -S tfm-enable-share-uart
```

See :ref:`cmake_options` for more details.

.. _tfm_enable_share_uart:

> Shared UART for application and TFM logging
> ====================================================

Reviewer: ```suggestion
Shared UART for application and TF-M logging
============================================
```

> If you intend to activate tfm logging while having modem traces enabled, it can be useful to direct the tfm logs to the same UART (UART0) as the
> application utilizes. Adding both the ``nrf91-modem-trace-uart`` and ``tfm-enable-share-uart`` snippets can be done in the following way:

Reviewer: ```suggestion
If you want to activate TF-M logging while having modem traces enabled, it can be useful to direct the TF-M logs to the UART (UART0) used by the application.
To activate both modem traces and TF-M logs, use the following command:
```

> .. code-block:: console
> 
>    west build --board <your_board> -S nrf91-modem-trace-uart -S tfm-enable-share-uart

Reviewer: ```suggestion
.. parsed-literal::
   :class: highlight

   west build --board *your_board* -S nrf91-modem-trace-uart -S tfm-enable-share-uart
```
